server/middlewares/authentication.coffee

```diff
@@ -50,19 +53,45 @@ module.exports.authenticate = (req, res, next) ->
             next error
         else if not user and user isnt undefined
             msg = localization.t "error otp invalid code"
-            error = new Error msg
-            error.status = 401
-            next error
-        else
             User.first (err, user) ->
-                req.logIn user, (err, info) ->
-                    if err
-                        msg = localization.t "error login failed"
-                        error = new Error msg
-                        error.status = 401
-                        next error
-                    else
-                        res.status(200).send success: true
+                recoveryCodes = user.encryptedRecoveryCodes[0]
+                if parseInt(req.body.authcode) in recoveryCodes
+                    # Disabling recovery code
+                    index = recoveryCodes.indexOf(parseInt req.body.authcode)
+                    recoveryCodes.splice index, 1
+                    console.log "------" + recoveryCodes.length
```

Comment:
Maybe we can safely remove the `console.log` call (except if it's intended for logging)